addons/mail/models/mail_channel.py

```diff
@@ -322,6 +322,8 @@ def action_unfollow(self):
         return self._action_unfollow(self.env.user.partner_id)
 
     def _action_unfollow(self, partner):
+        if partner not in self.channel_partner_ids:
```

Comment:
It probably means you cannot remove an archived partner, as channel_partner_ids should not contain archived records. Would probably require to check, maybe use channel partner model instead, and unit tests :) .

Reply:
After discussing it with @qho-odoo I guess the issue actually is that this method does not remove the follower. It's just removing the `channel.partner`. But the notification in v14 is sent based on followers, so it can keep being sent in a loop?

So I think we should also make the method remove follower?

But I think it's good to also keep the current condition (probably with active_test=False), because even if the follower issue wasn't there, it would make no sense to send the leave messages on calling the method on a non-member.

Reply:
What is exactly your use case ? Normally you never follow a channel, unless you go on the form view and click manually on "follow". But this use case is not really supported (and removed in master).

Reply:
This is the use case, there are DB with people that are followers on channels and it is making issues... (no kidding)

In this case they are using it on purpose to receive notifications on channels.

But we also added followers "incorrectly" ourselves since forever: when people were using mentions in channels, because of the lack of `mail_post_autofollow` that you saw I added only recently on another PR, every mentioned person at same point became a follower.

Reply:
Then call message_unsubscribe to be sure indeed. Still requesting an unit test :D .